swap_meet/vendor.py

```diff
@@ -1,2 +1,96 @@
+from queue import PriorityQueue
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory  = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            item not in self.inventory
+            return False
+    
+    def get_by_category(self,category):
+        category_list= []
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+    
+    def swap_items(self, other_vendor, my_item,their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        if my_item in self.inventory:
+            self.inventory.remove(my_item)
+            other_vendor.add(my_item)
+            self.inventory.append(their_item)
+            other_vendor.remove(their_item)
+            return True
+        
+#Wave 4
+    def swap_first_item(self, other_vendor):
+        if len(self.inventory) < 1 or len(other_vendor.inventory) < 1:
```

Comment:
This works, but it probably makes more sense to check whether the list length is 0 in both cases.